### DISCUSSION TITLE
Reminder: Silence a new cmake warning

### DISCUSSION THREAD

**Discussion by tsujan** [lxqt/lxqt#2461] [General]
It says something like this, depending on the project:

```
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.
```

I don't know though if it has come into all rolling distrors (here it's Arch/Manjaro).

**Comment by yan12125** [comment]
I'm fine with bumping the CMake requirement. CMake 3.6 is already 7 years old: https://cmake.org/files/v3.6/.

> I don't know though if it has come into all rolling distrors (here it's Arch/Manjaro).

It is deprecated by upstream: https://cmake.org/cmake/help/latest/release/3.27.html#deprecated-and-removed-features, so any distro with CMake >= 3.27 (https://repology.org/project/cmake/versions) will see this warning.

**Reply by tsujan** [reply]
In FeatherPad, I bumped it to 3.5.0. In LXQt, I follow the devs' decision; 3.6 seems OK.

**Reply by yan12125** [reply]
Oops, I misunderstand the warning. As 3.5 is still supported by the latest CMake, requiring 3.5 for LXQt is enough.

**Comment by redtide** [comment]
From my lxqt/lxqt-panel#2033

> the warning itself suggests the solution by adding a ...3.5 to the minimum version line, but I would set a recent version instead for all LXQt projects for consistence, like a 3.18 which is the one used in Debian Bullseye/oldstable.
I might submit a PR for all projects if defined.

**Reply by tsujan** [reply]
It will be silenced in all components before the next release.

For now, the focus is on porting LXQt to Qt6 and enhancing the Wayland support. No Qt5-based component will be released.

**Reply by redtide** [reply]
I'm aware about the current efforts, I just wanted to propose to use a newer version instead just silence it with a very old one.

**Reply by tsujan** [reply]
Yes, 3.18 seems quite acceptable.
